src/lib.rs

```diff
@@ -57,11 +58,37 @@ pub trait Number: Sized {
     /// Maximum value that can be represented by this type
     const MAX: Self;
 
+    /// Creates a number from the given value, throwing an error if the value is too large
     fn new(value: Self::UnderlyingType) -> Self;
 
+    /// Creates a number from the given value, return None if the value is too large
     fn try_new(value: Self::UnderlyingType) -> Result<Self, TryNewError>;
 
     fn value(self) -> Self::UnderlyingType;
+
+    #[cfg(not(feature = "const_convert_and_const_trait_impl"))]
+    fn new_<T: Number>(value: T) -> Self;
```

Comment:
What's the role of the trailing underscore here? Just to differentiate from `::new`?